stem/IO/kratos_solver_io.py

```diff
@@ -0,0 +1,220 @@
+        for model_part in model_parts:
+            if isinstance(model_part, BodyModelPart):
+                model_parts_dict["problem_domain_sub_model_part_list"].append(model_part.name)
+                model_parts_dict["body_domain_sub_model_part_list"].append(model_part.name)
+            else:
+                model_parts_dict["processes_sub_model_part_list"].append(model_part.name)
+
+        return model_parts_dict
+
+    def __create_solver_settings_dictionary(self, solver_settings: SolverSettings, mesh_file_name: str,
+                                            materials_file_name: str, model_parts: List[ModelPart]):
+        """
+        Creates a dictionary containing the solver settings
+
+        Args:
+            - solver_settings (:class:`stem.solver.SolverSettings`): The solver settings
+            - mesh_file_name (str): The name of the mesh file
+            - materials_file_name (str): The name of the materials file
+            - model_parts (List[:class:`stem.model_part.ModelPart`]): The list of model parts
+
+        Returns:
+            - Dict[str, Any]: dictionary containing the solver settings
+        """
+        solver_settings_dict: Dict[str, Any] = {"solver_type": "U_Pw",
+                                                "model_part_name": self.domain,
+                                                "domain_size": self.ndim,
+                                                "start_time": solver_settings.time_integration.start_time,
+                                                "model_import_settings": {
+                                                    "input_type": "mdpa",
+                                                    "input_filename": mesh_file_name},
+                                                "material_import_settings": {
+                                                    "materials_filename": materials_file_name},
+                                                "time_stepping": {
+                                                    "time_step": solver_settings.time_integration.delta_time,
+                                                    "max_delta_time_factor": solver_settings.time_integration.max_delta_time_factor},
+                                                "reduction_factor":  solver_settings.time_integration.reduction_factor,
+                                                "increase_factor": solver_settings.time_integration.increase_factor,
+                                                "buffer_size": 2,
+                                                "echo_level": 1,
+                                                "clear_storage": False,
+                                                "compute_reactions": False,
+                                                "move_mesh_flag": False,
+                                                "reform_dofs_at_each_step": False,
+                                                "nodal_smoothing": False,
```

Comment:
nodal smoothing means that all the nodes are the same? if so this should be an option

Reply:
i don't know what it means, I'll look into it

Reply:
it means that cauchy stress is extrapolated from gauss points to nodes. Should this still be an option?